tools/cldr-code/src/main/java/org/unicode/cldr/util/personname/PersonNameFormatter.java

```diff
@@ -292,10 +315,38 @@ public String format(NameObject nameObject) {
                 if (literal != null) {
                     result.append(literal);
                 } else {
-                    final String bestValue = nameObject.getBestValue(element.getModifiedField(), remainingModifers);
-                    if (!remainingModifers.isEmpty()) {
+                    String bestValue = nameObject.getBestValue(element.getModifiedField(), remainingModifers);
+                    if (bestValue == null) {
+                        // TODO Rich to handle collapsing whitespace / literals
+                        continue;
+                    } if (!remainingModifers.isEmpty()) {
                         // TODO Alex Apply unhandled modifiers algorithmically where possible
```

Comment:
I'm confused-- isn't this what you're doing below?

Reply:
Ok. I should delete that, since it is no longer applicable.